app/MMB/src/components/sectionListScrollTo/SectionListScrollTo.js

```diff
@@ -0,0 +1,240 @@
+// @flow
+
+import * as React from 'react';
+import { SectionList } from 'react-native';
+
+import OnLayout from './OnLayout';
+
+type Section = {|
+  +header: React.Node,
+  +data: React.Node[],
+  +shouldScroll?: boolean,
+|};
+
+type RenderItem = {|
+  +item: React.Node[],
+  +index: number,
+  +section: Section,
+|};
+
+type Props = {|
+  +data: Section[],
+|};
+
+/**
+ * SectionList considers headers, items and footers to get the layout
+ * /!\ Must not forget about footers with height 0
+ *
+ *  |---------------------------|
+ *  | HEADER = section.header
+ *  |---------------------------|
+ *  | Item1 = section.data[0]
+ *  |---------------------------|
+ *  | Item2 = section.data[1]
+ *  |---------------------------|
+ *  | ...
+ *  |---------------------------|
+ *  | FOOTER = null
+ *  |---------------------------|
+ *
+ */
+
+type ItemLayoutInfo = {|
+  height: number,
+  rendered: boolean,
+|};
+
+const generateDefaultLayoutInfo = (length: number): ItemLayoutInfo[] =>
+  [...Array(length)].map(() => ({
+    height: 0,
+    rendered: false,
+  }));
+
+const getLayoutInfoLength = (data: Section[]) =>
+  data.reduce((acc, section) => acc + 1 + section.data.length + 1, 0);
+
+export default class SectionListScrollTo extends React.Component<Props> {
+  layoutInfo: ItemLayoutInfo[] = generateDefaultLayoutInfo(
+    getLayoutInfoLength(this.props.data),
```

Comment:
if you are passing the method props, it should probably be a class method instead